### PR TITLE
fix: markdownlint documentation

### PR DESCRIPTION
**Pull request by sheeeng**
- Fix markdown files with [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) tool.
- ~Automate [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli) with GitHub Actions for consistent markdown content.~
- Fix usual markdown formatting issues.
- Remove unnecessary trailing whitespaces, etc.

### CODE REVIEW

File: website/docs/r/stream_analytics_output_servicebus_topic.html.markdown [hunk 1]
```diff
@@ -91,7 +91,7 @@ A `serialization` block supports the following:
 
 -> **NOTE:** This is required when `type` is set to `Csv` or `Json`.
 
-* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `   ` (tab), `|` (pipe) and `;`.
+* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are `` (space), `,` (comma), `` (tab), `|` (pipe) and `;`.
```
Comment (katbyte) [inline website/docs/r/stream_analytics_output_servicebus_topic.html.markdown:94]:
```suggestion
* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `    ` (tab), `|` (pipe) and `;`.
```

File: website/docs/r/stream_analytics_output_servicebus_queue.html.markdown [hunk 2]
```diff
@@ -90,7 +90,7 @@ A `serialization` block supports the following:
 
 -> **NOTE:** This is required when `type` is set to `Csv` or `Json`.
 
-* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `   ` (tab), `|` (pipe) and `;`.
+* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are `` (space), `,` (comma), `` (tab), `|` (pipe) and `;`.
```
Comment (katbyte) [inline website/docs/r/stream_analytics_output_servicebus_queue.html.markdown:93]:
```suggestion
* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `    ` (tab), `|` (pipe) and `;`.
```

File: website/docs/r/stream_analytics_output_eventhub.html.markdown [hunk 3]
```diff
@@ -90,7 +90,7 @@ A `serialization` block supports the following:
 
 -> **NOTE:** This is required when `type` is set to `Csv` or `Json`.
 
-* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `   ` (tab), `|` (pipe) and `;`.
+* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are `` (space), `,` (comma), `` (tab), `|` (pipe) and `;`.
```
Comment (katbyte) [inline website/docs/r/stream_analytics_output_eventhub.html.markdown:93]:
```suggestion
* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `    ` (tab), `|` (pipe) and `;`.
```

File: website/docs/r/stream_analytics_output_blob.html.markdown [hunk 4]
```diff
@@ -98,7 +98,7 @@ A `serialization` block supports the following:
 
 -> **NOTE:** This is required when `type` is set to `Csv` or `Json`.
 
-* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `   ` (tab), `|` (pipe) and `;`.
+* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are `` (space), `,` (comma), `` (tab), `|` (pipe) and `;`.
```
Comment (katbyte) [inline website/docs/r/stream_analytics_output_blob.html.markdown:101]:
```suggestion
* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are ` ` (space), `,` (comma), `    ` (tab), `|` (pipe) and `;`.
```

File: website/docs/r/public_ip.html.markdown [hunk 5]
```diff
@@ -42,7 +42,7 @@ The following arguments are supported:
 
 * `allocation_method` - (Required)  Defines the allocation method for this IP address. Possible values are `Static` or `Dynamic`.
 
-~> **Note** `Dynamic` Public IP Addresses aren't allocated until they're assigned to a resource (such as a Virtual Machine or a Load Balancer) by design within Azure - [more information is available below](#ip_address).
+~> **Note** `Dynamic` Public IP Addresses aren't allocated until they're assigned to a resource (such as a Virtual Machine or a Load Balancer) by design within Azure. See `ip_address` argument.
```
Comment (katbyte) [inline website/docs/r/public_ip.html.markdown:45]: this seems like an error?

File: website/docs/r/lb.html.markdown [hunk 6]
```diff
@@ -58,7 +58,7 @@ The following arguments are supported:
 -> **NOTE:** The `Microsoft.Network/AllowGatewayLoadBalancer` feature is required to be registered in order to use the `Gateway` SKU. The feature can only be registered by the Azure service team, please submit an [Azure support ticket](https://azure.microsoft.com/en-us/support/create-ticket/) for that.
 
 * `sku_tier` - (Optional) `sku_tier` - (Optional) The SKU tier of this Load Balancer. Possible values are `Global` and `Regional`. Defaults to `Regional`. Changing this forces a new resource to be created.
-* 
+*
```
Comment (katbyte) [inline website/docs/r/lb.html.markdown:61]:
```suggestion

```

File: website/docs/r/data_factory_pipeline.html.markdown [hunk 7]
```diff
@@ -41,16 +41,16 @@ resource "azurerm_data_factory_pipeline" "test" {
...
+ {
+  "name": "Append variable1",
+  "type": "AppendVariable",
+  "dependsOn": [],
+  "userProperties": [],
+  "typeProperties": {
+   "variableName": "bob",
+   "value": "something"
+  }
+ }
```
Comment (katbyte) [inline website/docs/r/data_factory_pipeline.html.markdown:53]: we probably want to keep the tabs or replace them with enough spaces to match the formatting closer?

File: website/docs/r/cosmosdb_mongo_collection.html.markdown [hunk 8]
```diff
@@ -51,11 +51,13 @@ The following arguments are supported:
 * `shard_key` - (Required) The name of the key to partition on for sharding. There must not be any other unique index keys.
 * `analytical_storage_ttl` - (Optional) The default time to live of Analytical Storage for this Mongo Collection. If present and the value is set to `-1`, it is equal to infinity, and items don’t expire by default. If present and the value is set to some number `n` – items will expire `n` seconds after their last modified time.
 * `default_ttl_seconds` - (Optional) The default Time To Live in seconds. If the value is `-1`, items are not automatically expired.
-* # `index` - (Optional) One or more `index` blocks as defined below.
+
+* `index` - (Optional) One or more `index` blocks as defined below.
+
```
Comment (katbyte) [inline website/docs/r/cosmosdb_mongo_collection.html.markdown:56]:
```suggestion
* `index` - (Optional) One or more `index` blocks as defined below.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Hi @sheeeng 

Thank you for the PR and all the markdown fixes! We will gladly accept the tidying - however i'm not sure we can accept the GHA. We like to have all the linting available while working locally, ie a script/go program that is called via the makefile. In this case it would be added to the `make website-lint` and just start doing its thing, so only having it in a GHA i'm afraid will not work for us, nor it being node as we neither run it locally/want to require contributors to have it.

While a go equivalent might work i'm not sure it would be able to run without error, there are a couple of instances i've left comments inline where i think the linter made a mistake. 

Either way if we can fix those up and remove the GHA i'll happily accept this - it cleans up so much so!

File: website/docs/r/public_ip.html.markdown (hunk 5 above)
Comment (sheeeng) [reply website/docs/r/public_ip.html.markdown:45]: The `#ip_address` link anchor is not a proper Markdown heading, thus markdownlint complains about it breaking `MD051/link-fragments` rule. This is the only instance in the documentation so far that had this unique way of linking to other argument.

How about the following sentence?

See `ip_address` argument below.

File: website/docs/r/stream_analytics_output_servicebus_queue.html.markdown (hunk 2 above)
Comment (sheeeng) [reply website/docs/r/stream_analytics_output_servicebus_queue.html.markdown:93]: Thank you for the feedback. The listing of possible values help users to copy pasting those given values. However, I don't see the benefits of listing the literal space and literal tab characters as the [original code](https://raw.githubusercontent.com/hashicorp/terraform-provider-azurerm/1ab58972d7891a5fa098fca0a3026068b67abb9d/website/docs/r/stream_analytics_output_servicebus_topic.html.markdown) will render them into unusable empty code block. Also, see below screenshot taken from the [generated documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/stream_analytics_output_servicebus_topic#field_delimiter) below. Both space and tab code block look exactly the same.
image


How about the following workaround? 

* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are (space character), `,` (comma character), (tab character), `|` (pipe character) and `;` (semicolon character).

File: website/docs/r/stream_analytics_output_eventhub.html.markdown (hunk 3 above)
Comment (sheeeng) [reply website/docs/r/stream_analytics_output_eventhub.html.markdown:93]: How about the following workaround? 

* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are (space character), `,` (comma character), (tab character), `|` (pipe character) and `;` (semicolon character).

File: website/docs/r/stream_analytics_output_blob.html.markdown (hunk 4 above)
Comment (sheeeng) [reply website/docs/r/stream_analytics_output_blob.html.markdown:101]: How about the following workaround with MD038/no-space-in-code?

* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are (space character), `,` (comma character), (tab character), `|` (pipe character) and `;` (semicolon character).

File: website/docs/r/data_factory_pipeline.html.markdown (hunk 7 above)
Comment (sheeeng) [reply website/docs/r/data_factory_pipeline.html.markdown]: The [original code](https://raw.githubusercontent.com/hashicorp/terraform-provider-azurerm/2daadb0f15ce7b64107c863ea76526b0b3c6f01a/website/docs/r/data_factory_pipeline.html.markdown) is using tab characters. Fixed the indentation with spaces with 3e8c1095f0ee184abbdaf6e0ea19cfdc55c15c78 commit. Is that okay?

**Comment by sheeeng** [comment]
Thanks for the feedback, @katbyte! I will work on improving `make website-lint` section.

> Hi @sheeeng
> 
> Thank you for the PR and all the markdown fixes! We will gladly accept the tidying - however i'm not sure we can accept the GHA. We like to have all the linting available while working locally, ie a script/go program that is called via the makefile. In this case it would be added to the `make website-lint` and just start doing its thing, so only having it in a GHA i'm afraid will not work for us, nor it being node as we neither run it locally/want to require contributors to have it.
> 
> While a go equivalent might work i'm not sure it would be able to run without error, there are a couple of instances i've left comments inline where i think the linter made a mistake.
> 
> Either way if we can fix those up and remove the GHA i'll happily accept this - it cleans up so much so!

File: website/docs/r/stream_analytics_output_servicebus_topic.html.markdown (hunk 1 above)
Comment (sheeeng) [reply website/docs/r/stream_analytics_output_servicebus_topic.html.markdown:94]: How about the following workaround with MD038/no-space-in-code?

* `field_delimiter` - (Optional) The delimiter that will be used to separate comma-separated value (CSV) records. Possible values are (space character), `,` (comma character), (tab character), `|` (pipe character) and `;` (semicolon character).

**Comment by sheeeng** [comment]
Implemented [markdownlint-cli in make website-lint](https://github.com/hashicorp/terraform-provider-azurerm/pull/18972/commits/81f39df0e1140bf6440492fc3ba94666a941f5a0).

Could you check that this workflow works well with existing workflows, @katbyte?

**Comment by katbyte** [comment]
@sheeeng - as i mentioned in my original review, we do not want to add a node dependency to our toolchain.

**Comment by sheeeng** [comment]
> @sheeeng - as i mentioned in my original review, we do not want to add a node dependency to our toolchain, so markdownlint-cli will not work for us.

Noted. The tool is removed by [0ca66af](https://github.com/hashicorp/terraform-provider-azurerm/pull/18972/commits/0ca66af37165b90e615f6141894665d55512597c) commit.

File: website/docs/r/public_ip.html.markdown (hunk 5 above)
Comment (katbyte) [reply website/docs/r/public_ip.html.markdown:45]: works for me!

File: website/docs/r/data_factory_pipeline.html.markdown (hunk 7 above)
Comment (katbyte) [reply website/docs/r/data_factory_pipeline.html.markdown]: yep thanks!

**Review by katbyte — APPROVED** [review]
Thanks @sheeeng - LGTM now 🐧

**Pull request merged**